metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java

```diff
@@ -364,12 +371,39 @@ private void updateGraphServiceDiff(
       log.debug("Adding edges: {}", additiveDifference);
       additiveDifference.forEach(_graphService::addEdge);
     }
+
+    // Then update existing edges
+    if (mergedEdges.size() > 0) {
+      log.debug("Updating edges: {}", mergedEdges);
+      mergedEdges.forEach(_graphService::upsertEdge);
+    }
+  }
+
+  private static List<Edge> getMergedEdges(final Set<Edge> oldEdgeSet, final Set<Edge> newEdgeSet) {
+    final Map<Integer, com.linkedin.metadata.graph.Edge> oldEdgesMap = oldEdgeSet
+        .stream()
+        .map(edge -> Pair.of(edge.hashCode(), edge))
```

Comment:
this could bite us later

Reply:
like surround it with a try/catch?

Or wdym?